### ISSUE TITLE
PythonSession should always set recording-id when acquiring global session

### ISSUE THREAD

**Issue by jleibs** [rerun-io/rerun#1792]
Full discussion here:
https://github.com/rerun-io/rerun/pull/1778#discussion_r1158659407

TLDR:
 - We currently initialize the RecordingId during python init.
 - Until python init has been run, the RecordingId is left with a value of `ZERO`
 - It is conceivable (though practically difficult) to end up in a situation where some message is send using this RecordingId
   - Note: I don't believe this is actually possible in pure python code, but could show up in an edge case of another rust-based python extension logging something during init.

I think the right solution is to change `python_session()` to take the `Python<_>` object as an argument, which would then allow us to always initialize the session correctly. However, this requires some case since it introduces new potential for deadlocks, in particular the kinds of problems resolved in: https://github.com/rerun-io/rerun/pull/843

**Comment by aedm** [comment]
This can be closed — the underlying design that caused it is gone.

`PythonSession` was sunset and replaced by `RecordingStream` (PR #1985, released 0.6.0). The store id is now built as `StoreId::new(kind, app_id, recording_id.unwrap_or_else(RecordingId::random))`, so a fresh random non-zero `RecordingId` (`rec_<uuid>`) is assigned eagerly whenever a recording/global session is acquired — there is no ZERO sentinel anywhere in the id design (`re_log_types::RecordingId::random`).

Verified on 0.33.0: two auto inits produce different non-zero ids, and explicit ids are honored:

```
init A (auto id): 78c5f06ea94743bc84e8af4f6685fed7
init B (auto id): 96aa46d587d54698a21376e251b42827
PASS: two auto inits produced DIFFERENT random ids
PASS: explicit recording id honored
```
